contracts/BasicProvenance.sol

```diff
@@ -1,77 +1,62 @@
-pragma solidity ^0.4.20;
-contract WorkbenchBase {
-    event WorkbenchContractCreated(string applicationName, string workflowName, address originatingAddress);
-    event WorkbenchContractUpdated(string applicationName, string workflowName, string action, address originatingAddress);
-
-    string internal ApplicationName;
-    string internal WorkflowName;
-
-    function WorkbenchBase(string applicationName, string workflowName) internal {
-        ApplicationName = applicationName;
-        WorkflowName = workflowName;
-    }
-
-    function ContractCreated() internal {
-        WorkbenchContractCreated(ApplicationName, WorkflowName, msg.sender);
-    }
-
-    function ContractUpdated(string action) internal {
-        WorkbenchContractUpdated(ApplicationName, WorkflowName, action, msg.sender);
-    }
-}
-
-contract BasicProvenance is WorkbenchBase('BasicProvenance', 'BasicProvenance')
-{
+pragma solidity ^0.5.0;
 
+contract BasicProvenance {
     //Set of States
 	enum StateType { Created, InTransit, Completed}
-	
 	//List of properties
 	StateType public  State;
 	address public  InitiatingCounterparty;
 	address public  Counterparty;
 	address public  PreviousCounterparty;
 	address public  SupplyChainOwner;
 	address public  SupplyChainObserver;
-	
-	function BasicProvenance(address supplyChainOwner, address supplyChainObserver) public
-	{
+
+    event LogContractCreated(string applicationName, string workflowName, address originatingAddress);
+    event LogContractUpdated(string applicationName, string workflowName, string action, address originatingAddress);
+
+    string internal ApplicationName;
+    string internal WorkflowName;
+
+    function ContractCreated() private {
```

Comment:
Is there a need for this function when the calling function can emit the event directly?